### PR TITLE
PE-870: BATS for download drive

### PR DESCRIPTION
**Pull request by matibat**

### CODE REVIEW

File: bats_test/downloads/drive-downloads/download_drive.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    UNSAFE_PASSWORD="${4}"
+    DRIVE_KEY="${5}"
+
+    ARDRIVE_CMD="yarn ardrive download-drive \
+        $([ "${DRIVE_ID}" != "" ] && echo "-d ${DRIVE_ID}" )
+        $([ "${LOCAL_PATH}" != "" ] && echo "--local-path=${LOCAL_PATH}") \
+        $([ "${MAX_DEPTH}" != "" ] && echo "--max-depth=${MAX_DEPTH}") \
+        $(
+            [ "${UNSAFE_PASSWORD}" != "" ] && echo "--unsafe-drive-password=${UNSAFE_PASSWORD}" || 
+            [ "${DRIVE_KEY}" != "" ] && echo "--drive-key=${DRIVE_KEY}" || echo "-P"
```
Comment (matibat) [inline bats_test/downloads/drive-downloads/download_drive.sh:30]:
```suggestion
            [ "${DRIVE_KEY}" != "" ] && echo "--drive-key=${DRIVE_KEY}" ||
            echo "-P"
```

File: bats_test/downloads/drive-downloads/happy_paths.bats [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    assert_file_exist "${DIR}/Files With a Variety of Extensions Public/Coffee_chart.jpg"
+    assert_file_exist "${DIR}/Files With a Variety of Extensions Public/samplee.mkv"
+    # Skipped to avoid the wrong path error
+    # assert_file_exist "${DIR}/Files With a Variety of Extensions Public/test!_·$%d_%&·_.txt"
+
+    assert_file_exist "${DIR}/Files With a Variety of Extensions Public/CaSe/100byts.txt"
+    assert_file_exist "${DIR}/Files With a Variety of Extensions Public/CaSe/100bYtS.txt"
+}
+
+@test "Downloads the root folder of the drive into the dirname of the path if it's a non-existent item inside a valid directory" {
```
Comment (agsuy) [inline bats_test/downloads/drive-downloads/happy_paths.bats:41]: _"Download drive to a custom folder using a custom depth of 0"_ ??

File: bats_test/downloads/drive-downloads/cleanup.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+DIR="$(cd "$(dirname "${BATS_TEST_FILENAME}")" >/dev/null 2>&1 && pwd)"
+
+rm -rf \
+    "${DIR}/Files With a Variety of Extensions Public" \
+    "${DIR}/My custom folder name" \
+    "${DIR}/existing_file.txt" \
+    "${DIR}/existing_fifo"
```
Comment (matibat) [inline bats_test/downloads/drive-downloads/cleanup.sh:9]: We ain't using any FIFO in these tests
```suggestion
    "${DIR}/existing_file.txt"
```

File: bats_test/downloads/drive-downloads/happy_paths.bats (hunk 2 above)
Comment (matibat) [reply bats_test/downloads/drive-downloads/happy_paths.bats:41]:
```suggestion
@test "Download drive to a custom folder using a custom depth of 0" {
```

**Review by fedellen — APPROVED** [review]
🦇

**Pull request merged**